nucypher/characters/lawful.py

```diff
@@ -1748,34 +1735,27 @@ def interface_info_with_metadata(self):
     #
 
     def verify_kfrag_authorization(self,
-                                   alice: Alice,
-                                   kfrag: KeyFrag,
-                                   signed_writ: bytes,
-                                   work_order: 'WorkOrder'
+                                   hrac: bytes,
+                                   publisher: Alice,
+                                   authorized_kfrag: AuthorizedKeyFrag,
                                    ) -> VerifiedKeyFrag:
-        from nucypher.policy.orders import WorkOrder  # TODO: resolve ciruclar dependency
 
-        writ_hrac, writ_kfrag_checksum, writ_signature = WorkOrder.signed_writ_splitter(signed_writ)
-        reconstructed_writ = writ_hrac + writ_kfrag_checksum
+        # TODO: should it be a method of AuthorizedKeyFrag?
 
         try:
-            self.verify_from(alice, reconstructed_writ, signature=writ_signature)
+            self.verify_from(publisher, authorized_kfrag.writ, signature=authorized_kfrag.writ_signature)
         except InvalidSignature:
-            raise Policy.Unauthorized  # This isn't from Alice (relayer).
+            raise Policy.Unauthorized  # This isn't from Alice (publisher).
```

Comment:
Wanted this included somewhere in PR #2687 but it wasn't before getting merged - https://github.com/nucypher/nucypher/pull/2687#discussion_r664046936. Piggy-backing off of your PR 😅  
```suggestion
            # TODO differentiate cases for Policy.Unauthorized - #2740
            raise Policy.Unauthorized  # This isn't from Alice (publisher).
```